libwarpaffine/dowarp.cpp

```diff
@@ -55,12 +56,17 @@ DoWarp::OutputBrickInfoRepository::OutputBrickInfoRepository(const AppContext& c
     //  as the key.
     map<int, int> scene_index_to_m_index;
 
+    int currentZ = -1;
     for (const auto& item : document_info.map_brickid_position)
     { 
         Eigen::Vector3d edge_point;
         Eigen::Vector3d extent;
         DeskewHelpers::CalculateAxisAlignedBoundingBox(item.second.width, item.second.height, document_info.depth, transformation_matrix, edge_point, extent);
         DestinationBrickInfo destination_brick_info;
+        int slice_id = 0;
+
+        // new ID per Z
+        destination_brick_info.slice_id = slice_id++;
```

Comment:
I cannot really follow here - to my understanding, destination_brick_info.slice_id will always have the value 0. Is this as intended?

Reply:
it's a mistake, the slice_id should be initialized outside the loop, and we do not need the ´currentZ´ at all.